src/openforms/formio/typing/custom.py

```diff
@@ -9,5 +9,19 @@ class DateComponent(Component):
     customOptions: NotRequired[DatePickerCustomOptions]
 
 
+class ComponentValidation(TypedDict):
+    validate: dict
+    translatedErrors: dict
+
+
+class AddressValidationComponents(TypedDict):
+    postcode: NotRequired[ComponentValidation]
+    city: NotRequired[ComponentValidation]
+
+
+class CustomOpenFormsComponent(TypedDict):
+    components: AddressValidationComponents
+
+
 class AddressNLComponent(Component):
     deriveAddress: bool
```

Comment:
I'm not a big fan of bare `dict`'s as they don't give any additional useful information, especially when we know the shape upfront.

I also don't think you have `CustomOpenFormsComponent` hooked up anywhere? It might not be possible due to the inheritance rules of TypedDict. The type annotations/descriptions of formio components kinda suck, `TypedDict` isn't really flexible enough IMO.

Reply:
Oh whoops, I changed it and added it to the `AddressNLComponent`

Reply:
I'm not sure what is best to do here, since `AddressNLComponent` inherits from `Component` it already has `openForms` defined as `OpenFormConfig`, but I'm not sure if it makes sense (or even is possible) to define something like `AddressValidationComponents` in a generic way?